primitives/src/v2/mod.rs

```diff
@@ -1582,6 +1612,66 @@ pub fn supermajority_threshold(n: usize) -> usize {
 	n - byzantine_threshold(n)
 }
 
+/// Vector indexed via key value type e.g. ValidatorIndex
+#[derive(Clone, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(PartialEq))]
+pub struct TypeVec<K: From<usize>, V: Clone>(TiVec<K, V>);
+
+impl<K: From<usize>, V: Clone> TypeVec<K, V> {
+	///Creates new TypeVec object
+	pub fn new() -> Self {
+		TypeVec::<K, V> { 0: TiVec::<K, V>::new() }
+	}
+}
+
+impl<K: From<usize>, V: Clone> From<Vec<V>> for TypeVec<K, V> {
+	fn from(vec: Vec<V>) -> Self {
+		let mut type_vec: TypeVec<K, V> = TypeVec::new();
+		for value in vec.into_iter() {
+			type_vec.0.push(value)
+		}
+		type_vec
+	}
+}
+
+#[cfg(feature = "std")]
+impl<K: From<usize>, V: Clone> MallocSizeOf for TypeVec<K, V> {
+	fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
+		0
+	}
+	fn constant_size() -> Option<usize> {
+		Some(0)
+	}
+}
+
+impl<K: From<usize>, V: Clone> Deref for TypeVec<K, V> {
+	type Target = Vec<V>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0.raw
+	}
+}
+
+impl<K: From<usize>, V: Clone> DerefMut for TypeVec<K, V> {
+	fn deref_mut(&mut self) -> &mut Self::Target {
+		&mut self.0.raw
+	}
+}
+
+impl<K: From<usize>, V: Clone> WrapperTypeEncode for TypeVec<K, V> {}
+
+impl<K: From<usize>, V: Clone> WrapperTypeDecode for TypeVec<K, V> {
+	type Wrapped = Vec<V>;
+}
+
+impl<K: From<usize>, V: Clone> TypeInfo for TypeVec<K, V> {
+	type Identity = ();
+
+	fn type_info() -> Type {
+		todo!()
```

Comment:
nit: Must be fixed before merge

Reply:
I'm not quite sure how to implement this trait.